### PR TITLE
[WIP] Adding CTR metric

### LINKED ISSUE #27: Add CTR metric
### 🚀 The feature

**Objective**   
We want to implement a functional metric and class metric to calculate CTR (click through rate) based on inputs
CTR = sum( I(click) * weight) / sum(weights)

Examples:   

input: torch.tensor([0, 1, 0, 1, 1, 0, 0, 1])
weight: 1.0 (default value)
ctr: torch.tensor(0.5)

input: torch.tensor([0, 1, 0, 1, 1, 0, 0, 1])
weight: torch.tensor([1.0, 2.0, 1.0, 2.0, 1.0, 2.0, 1.0, 2.0])
ctr: torch.tensor(0.58333)

For multi-tasks situation
input: torch.tensor([[0, 1, 0, 1], [1, 0, 0, 1]])
weight: 1.0 (default value)
ctr: torch.tensor([0.5, 0.05])

### Motivation, pitch

Click-through rates are important for evaluating model performance and a commonly expected out-of-the-box metric someone might want to get started with.

### Alternatives

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by dbish**
Please read through our [contribution guide](https://github.com/pytorch-labs/torcheval/blob/main/CONTRIBUTING.md) prior to creating your pull request.

Summary:
1. ran pre-commit and fixed some changes before I added the metric (first commit)

Test plan:

Fixes #27

### CODE REVIEW

File: torcheval/metrics/toolkit.py [hunk 1]
```diff
@@ -11,11 +11,11 @@
 import torch
 import torch.distributed as dist
 
-from torcheval.metrics import Metric
```
Comment (DuYicong515) [inline torcheval/metrics/toolkit.py:14]: The file reorder is different from internal meta lint -- since here usort treated torcheval as a first-party library where internal usort will ignore this first-party detection.

Let's add a project.toml and ignore it to match the 2 behaviours

https://github.com/facebookresearch/recipes/blob/bc3c7e9424a8ab6444917b23098061abbb3205b7/pyproject.toml#L1-L2

File: tests/metrics/classification/test_precision_recall_curve.py [hunk 2]
```diff
@@ -4,11 +4,10 @@
 # This source code is licensed under the BSD-style license found in the
 # LICENSE file in the root directory of this source tree.
 
-from typing import List, Optional
+from typing import Optional
 
 import torch
 
-from sklearn.metrics import precision_recall_curve
```
Comment (ninginthecloud) [inline tests/metrics/classification/test_precision_recall_curve.py:11]: Thanks for catching this error! Let's have a rebase to make sure we don't have code change in unrelated files.
